pctasks/core/pctasks/core/utils/summary.py

```diff
@@ -0,0 +1,673 @@
+"""summary.py
+
+Module for generating summaries from JSON objects.
+Useful for generating Collection "summaries" and "item-assets" fields.
+"""
+
+
+from abc import abstractmethod
+from enum import Enum
+from functools import reduce
+from typing import Any, Dict, List, Optional, Set, Union, cast
+
+from pydantic import BaseModel, Field
+
+from pctasks.core.utils import map_opt
+
+
+class SummarySettings(BaseModel):
+    max_mixed_summary_samples = 4
+    max_mixed_value_samples = 4
+    max_mixed_object_list_samples = 3
+    max_mixed_key_sets = 5
+
+
+class ValueTypes(str, Enum):
+    BOOLEAN = "boolean"
+    INT = "numeric-int"
+    FLOAT = "numeric-float"
+    STRING = "string"
+    LIST = "list"
+    NULL = "null"
+
+
+class SummaryTypes(str, Enum):
+    DISTINCT = "distinct"
+    INT_RANGE = "int-range"
+    FLOAT_RANGE = "float-range"
+    OBJECT_LIST = "object-list"
+    MIXED_VALUE = "mixed-value"
+    MIXED_OBJECT_LIST = "mixed-object-list"
+    MIXED_SUMMARIES = "mixed-summaries"
+
+
+def is_value_type(v: Any) -> bool:
+    return (
+        isinstance(v, bool)
+        or isinstance(v, str)
+        or isinstance(v, float)
+        or isinstance(v, int)
+    )
```

Comment:
The second argument to isinstance can be a tuple of types.

```suggestion
    return isinstance(v, (bool, str, float, int))
```

Reply:
Nice, thanks!